prime-angular-frontend/src/app/modules/enrolment/shared/services/enrolment-form-state.service.ts

```diff
@@ -589,6 +592,49 @@ export class EnrolmentFormStateService extends AbstractFormStateService<Enrolmen
     });
   }
 
+  public addNonHealthAuthorityOboSite(siteForm: FormGroup, siteFormList: FormArray) {
+    const siteName = siteForm.get('siteName') as FormControl;
+    this.formUtilsService.setValidators(siteName, [Validators.required]);
+    siteFormList.push(siteForm);
+  }
+
+  /**
+   * @param haSiteForm - aka Health Authority Facility Form
+   * @param healthAuthoritySites - a FormArray where each element, representing a Health Authority where the enrollee works, contains a FormArray.
+   *  This nested FormArray contains a FormGroup for each facility that the enrollee works at, in that Health Authority
+   * @param healthAuthorityCode
+   */
+  public addHealthAuthorityOboSite(haSiteForm: FormGroup, healthAuthoritySites: FormGroup, healthAuthorityCode: number) {
+    const facilityName = haSiteForm.get('facilityName') as FormControl;
+    this.formUtilsService.setValidators(facilityName, [Validators.required]);
+    let sitesOfHealthAuthority = healthAuthoritySites.get(String(healthAuthorityCode)) as FormArray;
+    if (!sitesOfHealthAuthority) {
+      sitesOfHealthAuthority = this.fb.array([]);
+      sitesOfHealthAuthority.setValidators([FormArrayValidators.atLeast(1)]);
+      healthAuthoritySites.setControl(String(healthAuthorityCode), sitesOfHealthAuthority);
+    }
+    sitesOfHealthAuthority.push(haSiteForm);
+  }
+
+  public removeUnselectedHAOboSites() {
+    // Obo Sites need to be removed from two different collections
+    const oboSites = this.jobsForm.get('oboSites') as FormArray;
+    const healthAuthoritySites = this.jobsForm.get('healthAuthoritySites') as FormGroup;
+    const enrolleeHealthAuthorities = this.careSettingsForm.get('enrolleeHealthAuthorities') as FormArray;
+    // If the checkbox for the health authority is not selected, remove the corresponding Obo Sites
+    this.configService.healthAuthorities.forEach((healthAuthority, index) => {
+      if (!enrolleeHealthAuthorities.at(index).value) {
+        for (let i = oboSites.controls.length - 1; i >= 0; i--) {
```

Comment:
With JS being functional there isn't much need to do full `for` loop syntax and instead you can use `forEach` - `oboSites.controls.forEach((os: FormGroup) => { ... });`